[16.04] Provide backward compatible galaxy.datatypes.checkers module...

jmchilton opened the pull request:
... for external datatypes.

xref http://dev.list.galaxyproject.org/STACKS-error-td4669428.html

jmchilton committed:
Provide backward-compatible galaxy.datatypes.checkers module.

xref http://dev.list.galaxyproject.org/STACKS-error-td4669428.html

nsoranzo commented on lib/galaxy/datatypes/checkers.py:
```diff
@@ -0,0 +1,27 @@
+"""Module proxies :module:`galaxy.util.checkers` for backward compatiblity.
```
s/compatiblity/compatibility/

jmchilton commented:
@nsoranzo You reviewed this earlier - any chance I can get you to +1 it and merge?

nsoranzo commented on lib/galaxy/datatypes/checkers.py:
```diff
@@ -0,0 +1,27 @@
+    'check_html',
+    'check_image',
+    'check_zip',
+    'HTML_CHECK_LINES',
```
Not sure `HTML_CHECK_LINES` should be exported.

martenson commented:
why did Travis fail to run this?

martenson commented:
runs here
https://travis-ci.org/galaxyproject/galaxy/builds/132283289

dannon commented:
@jmchilton Did you want to address the HTML_CHECK_LINES point prior to a merge?

Otherwise, this looks fine to me as-is; exposing HTML_CHECK_LINES won't _break_ anything, though I hope no consumer tools are actually using that.

jmchilton commented:
I'll revise this in order to not expose that - the impulse of limiting what is exposed is good.

dannon commented:
@jmchilton I agree with the sentiment, but if this is only being added back for compatibility, it should expose exactly what was there before the move to util.compat, right?  (which was everything)

jmchilton committed:
Remove HTML_CHECK_LINES from checkers export.

As suggested by @nsoranzo and @dannon, we will restore if and only if someone finds a datatype that uses it.

jmchilton commented:
@dannon Well we are defining the interface we providing backward compatibility as we go here. Certainly I'd prefer if none of this stuff had been used and datatypes didn't even exist in the tool shed - at least not without a defined and maintainable interface to Galaxy.  But the pragmatic reality is they did exist - we shouldn't provide backward compatibility to more than we need to in order to fix actual exported datatypes though in my opinion.

dannon commented:
@jmchilton Hrmm, was looking at this more like a 16.04 bugfix, but since it's just HTML_CHECK_LINES it's not worth making the case for -- seems like this crept a bit, to me, though.

nsoranzo commented:
:+1: Thanks @jmchilton!
